crates/sui-graphql-rpc/src/types/big_int.rs

```diff
@@ -41,11 +22,43 @@ impl ScalarType for BigInt {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+struct NotANumber(String);
+
+impl Display for NotANumber {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.0)
+    }
+}
+
 impl FromStr for BigInt {
-    type Err = ();
+    type Err = InputValueError<BigInt>;
```

Comment:
This should be `NotANumber` -- reason being: `FromStr` may be used for reasons other than parsing input (that's why we abstracted it out), so you want its implementation to be agnostic.  `InputValueError` should still be produced by `ScalarType::parse`, which is implemented by calling `map_err` to interpret `NotANumber` as an `InputValueError`.